deploy/docs/Best_Practices.md

```diff
@@ -0,0 +1,68 @@
+
+# Advanced Configuration / Best Practices
+
+
+### Multiline Log Support
+
+By default, we use a regex that matches the first line of a multiline log starting with a date of the format : `2019-11-17 07:14:12`.
+
+Users can specify their custom regex to detect first line of a multiline log and parse the logs correctly.
+
+New parsers can be defined under the `parsers` key of the fluent-bit configuration section in `values.yaml` as follows:
+
+```
+parsers:
+  enabled: true
+  regex:
+    - name: new_parser_name
+    regex: (?<log>^{"log":"\d{4}-\d{1,2}-\d{1,2} \d{2}:\d{2}:\d{2}.*)
+```
+
+The regex needs to have at least one named capture group.
+
+In order to use the newly define parser, change the `Parser_Firstline` parameter in the `Input plugin` configuration of fluent-bit:
+
+```bash
+Parser_Firstline new_parser_name
+```
+
+
+### Fluentd File-based buffer
+
+By default, we use the in-memory buffer for the Fluentd buffer, but for production environments we recommend users to use the file-based buffer instead.
+
+Buffer configuration can be set in `values.yaml` under the `sumologic` key as follows:
+
+```
+fluentd:
+## Option to specify the Fluentd buffer as file/memory.
+   buffer: "file"
+```
+
+Additional buffering and flushing parameters can be added in the `buffer.output.conf`config file, to be used by all buffer sections.
+
+We recommend the following configurations for the file-based buffer:
+
+#### Buffering parameters:
+```
+flush_interval 5s
+
+flush_thread_count 4
+
+chunk_limit_size 100k
+
+total_limit_size 128m
+```
+
+####  Flushing parameters:
+```
+retry_wait 5
+
+retry_forever false
```

Comment:
we recommend dropping data? for production environments this seems like a bad idea.  We should recommend sizing the volume appropriately to not drop data.

Reply:
For  `retry_forever, If `true`, plugin will ignore `retry_timeout` and `retry_max_times` options and retry flushing forever. 

There is another buffering parameter which defines how output plugin behaves when its buffer queue is full.
Default : throw_exception
`overflow_action` [enum:
throw_exception/block/drop_oldest_chunk]